sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -516,6 +516,7 @@ case class NewInstance(
 
   override def nullable: Boolean = needNullCheck
 
+  override def foldable: Boolean = false
```

Comment:
Alternatively, I could make it false only when there are no arguments (i.e., `children.size` is zero).